utils/utils.go

```diff
@@ -139,3 +152,65 @@ func GetRouteReference(r kong.Route) *kong.Route {
 	}
 	return route
 }
+
+// ParseKongVersion takes a version string from the Gateway and
+// turns it into a semver-compliant version to be used for
+// comparison across the code.
+func ParseKongVersion(version string) (semver.Version, error) {
+	v, err := CleanKongVersion(version)
+	if err != nil {
+		return semver.Version{}, err
+	}
+	return semver.ParseTolerant(v)
+}
```

Comment:
Heavily unit test this function.

Reply:
https://github.com/Kong/deck/pull/753/commits/e5d3912bd1141f0fec3d4694bd8272942862aa8d